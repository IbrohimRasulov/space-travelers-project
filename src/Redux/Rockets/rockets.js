const GET_ROCKETS_API = 'SpaceTravelersHub/missions/GET_ROCKETS_API';
const ROCKET_API = 'https://api.spacexdata.com/v3/rockets';

const initialState = [];

export const getRockets = (payload) => ({
  type: GET_ROCKETS_API,
  payload,
});

export const getRocketsFromAPI = () => async (dispatch) => {
  await fetch(`${ROCKET_API}`)
    .then((response) => response.json())
    .then((data) => {
      const arrData = data.map((rocket) => ({
        id: rocket.id,
        name: rocket.rocket_name,
        description: rocket.description,
        image: rocket.flickr_images[0],
        reserved: false,
      }));
      if (arrData) {
        dispatch(getRockets(arrData));
      }
    });
};

const rocketsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_ROCKETS_API:
      return [...action.payload];
    default:
      return state;
  }
};

export default rocketsReducer;
