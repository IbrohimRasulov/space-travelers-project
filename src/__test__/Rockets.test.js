import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/configureStore';
import Rockets from '../components/Rockets';

it('testing rendering Rockets component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Rockets />
      </Router>
    </Provider>,
  ).toJSON;
  expect(tree).toMatchSnapshot();
});
