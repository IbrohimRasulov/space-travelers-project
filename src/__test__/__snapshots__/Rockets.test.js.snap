// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing rendering Rockets component 1`] = `[Function]`;
